azure/scope/machine.go

```diff
@@ -382,6 +385,40 @@ func (m *MachineScope) SetFailureReason(v capierrors.MachineStatusError) {
 	m.AzureMachine.Status.FailureReason = &v
 }
 
+// SetCondition sets a condition on the AzureMachine.
+func (m *MachineScope) SetCondition(condition clusterv1.ConditionType, reason string, severity clusterv1.ConditionSeverity, value bool) {
+	if value {
+		conditions.MarkTrue(m.AzureMachine, condition)
+	} else {
+		conditions.MarkFalse(m.AzureMachine, condition, reason, severity, "")
+	}
+}
+
+func (m *MachineScope) UpdateStatus() {
+	switch m.VMState() {
+	case infrav1.VMStateSucceeded:
```

Comment:
I like the idea of movie this logic to scope. Just a side note not related to the pr, as we add more stuff to scope, we may want to prioritize #1108